pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -43,43 +42,81 @@ import (
 	"istio.io/istio/pkg/spiffe"
 	"istio.io/istio/pkg/util/sets"
 	"istio.io/istio/pkg/workloadapi"
+	"istio.io/istio/pkg/workloadapi/security"
 )
 
 // AmbientIndex maintains an index of ambient WorkloadInfo objects by various keys.
 // These are intentionally pre-computed based on events such that lookups are efficient.
 type AmbientIndex struct {
 	mu sync.RWMutex
-	// byService indexes by Service (virtual) *IP address*. A given Service may have multiple IPs, thus
+	// byService indexes by network/Service (virtual) *IP address*. A given Service may have multiple IPs, thus
 	// multiple entries in the map. A given IP can have many workloads associated.
-	byService map[string][]*model.WorkloadInfo
-	// byPod indexes by Pod IP address.
+	byService map[networkAddress][]*model.WorkloadInfo
+	// byPod indexes by network/podIP address.
 	byPod map[string]*model.WorkloadInfo
```

Comment:
I thought we were going to change this to `networkAddress` internal struct?

Reply:
done